### DISCUSSION TITLE
Can filters stop a user query and provide a reply?

### DISCUSSION THREAD

**Discussion by PlebeiusGaragicus** [open-webui/open-webui#3724] [Q&A] [answered]
Here's an example use case - / SLASH commands similar to how Telegram bots work.

For example, if I have a bot that accepts micropayments for token-usage I may want to type "/BAL" to see what my account balance is.  Or, I may want to top up my account with "/PAY" and have it generate an invoice.  The sky is the limit here with what you can do.  If my query doesn't start with a '/' then I'll run it through the LLM chain.

But, my question is if filters can catch these slash commands and reply directly to the user instead of simply modifying the body response.  If so, I haven't been able to find an example in code.

**Comment by andreybond** [comment, marked as answer]
I think You can try to implement this via pipelines, but you will not get the type hints which would be very nice.

**Reply by PlebeiusGaragicus** [reply]
Agreed - after further research it seems like using a Pipeline is the way to go here.
